DilemmaGX/Animator.js

```diff
@@ -0,0 +1,773 @@
+                    },
+                    {
+                        text: this.formatMessage('Ani.easeInBack'),
+                        value: 'easeInBack'
+                    },
+                    {
+                        text: this.formatMessage('Ani.easeOutBack'),
+                        value: 'easeOutBack'
+                    },
+                    {
+                        text: this.formatMessage('Ani.easeInOutBack'),
+                        value: 'easeInOutBack'
+                    },
+                    {
+                        text: this.formatMessage('Ani.easeInBounce'),
+                        value: 'easeInBounce'
+                    },
+                    {
+                        text: this.formatMessage('Ani.easeOutBounce'),
+                        value: 'easeOutBounce'
+                    },
+                    {
+                        text: this.formatMessage('Ani.easeInOutBounce'),
+                        value: 'easeInOutBounce'
+                    }
+                ],
+                Effect: [
+                    {
+                        text: this.formatMessage('Ani.color'),
+                        value: 'color'
+                    },
+                    {
+                        text: this.formatMessage('Ani.fisheye'),
+                        value: 'fisheye'
+                    },
+                    {
+                        text: this.formatMessage('Ani.whirl'),
+                        value: 'whirl'
+                    },
+                    {
+                        text: this.formatMessage('Ani.pixelate'),
+                        value: 'pixelate'
+                    },
+                    {
+                        text: this.formatMessage('Ani.mosaic'),
+                        value: 'mosaic'
+                    },
+                    {
+                        text: this.formatMessage('Ani.brightness'),
+                        value: 'brightness'
+                    },
+                    {
+                        text: this.formatMessage('Ani.ghost'),
+                        value: 'ghost'
+                    },
+                ]
+            }
+        }
+    }
+    //Defines start here
+
+    docs() {
+        console.log("Opening Animator.js documentation in new window...");
+        window.open("https://gandi.rth1.app/Docs/Animator.js.html");
+    }
+
+    //value return
+    //note that for advanced animation involving movements, resizing, effects, or directions, this block would be better than the Move() function.
+    Animate(args) {
+        //fix input values (a logic related bug, but this can fix it)
+        let val_2 = args.val_2 - args.val_1
+
+        return Ani_Tweaks(args.menu, args.ini_t, args.val_1, val_2, args.duration);//val_2 not args.val2 is ment to fix the logic bug mentioned before.
+    }
+
+    //move (frame animation)
+    Move(args, util) {
+        let x_2 = args.x_2 - args.x_1;
+        let y_2 = args.y_2 - args.y_1;
```

Comment:
Here, for args.x_1 and y_1 is just the sprite's position, you can use util.target.x and util.target.y instead

Reply:
Even though util.target.x and util.target.y is a great method, for these blocks, they only move the characters to the position of a frame of the entire animation. So, a better solution is to inform the extension where the animation really begins. However, the animated glideTo block is still on the way. For a more up-to-date version, please use this extension (Gandi only, insider version reserved.): https://gandi.rth1.app/Scripts/Animator_beta.js